src/vs/workbench/contrib/terminal/browser/xterm/xtermTerminal.ts

```diff
@@ -608,6 +631,20 @@ export class XtermTerminal extends DisposableStore implements IXtermTerminal, II
 		};
 	}
 
+	private async _enableLigatures(): Promise<void> {
+		if (!this.raw.element || this._ligaturesAddon) {
```

Comment:
where does the `element` get attached to `raw` and when this happens, will we attempt to enable ligatures later or accept that it couldn't happen?

Reply:
`element` is set when `Terminal.open` is called. This is fine since we call `_enableLigatures` again in `attachToElement`